internal/nodes/avi_model_evh_nodes.go

```diff
@@ -740,77 +740,83 @@ func ProcessInsecureHostsForEVH(routeIgrObj RouteIngressModel, key string, parse
-				Name:         evhNodeName,
-				VHParentName: vsNode[0].Name,
-				Tenant:       lib.GetTenant(),
-				EVHParent:    false,
-				EvhHostName:  host,
-				ServiceMetadata: avicache.ServiceMetadataObj{
-					NamespaceIngressName: ingressHostMap.GetIngressesForHostName(host),
-					Namespace:            namespace,
-					HostNames:            hostSlice,
-				},
-			}
+	utils.AviLog.Debugf("key: %s, msg: Storedhosts after processing insecurehosts: %s", key, utils.Stringify(Storedhosts))
+}
 
-			foundEvhModel := FindAndReplaceEvhInModel(evhNode, vsNode, key)
-			if !foundEvhModel {
-				vsNode[0].EvhNodes = append(vsNode[0].EvhNodes, evhNode)
-			}
-		} else {
-			// The evh node exists, just update the svc metadata
-			evhNode.ServiceMetadata.NamespaceIngressName = ingressHostMap.GetIngressesForHostName(host)
-			evhNode.ServiceMetadata.Namespace = namespace
-			evhNode.ServiceMetadata.HostNames = hostSlice
-		}
-		evhNode.ServiceEngineGroup = lib.GetSEGName()
-		evhNode.VrfContext = lib.GetVrf()
-		evhNode.ApplicationProfile = utils.DEFAULT_L7_APP_PROFILE
-		// Remove the redirect for secure to insecure transition
-		hosts := []string{host}
-		if pathsvcmap.gslbHostHeader != "" {
-			hosts = append(hosts, pathsvcmap.gslbHostHeader)
+func (o *AviObjectGraph) BuildModelGraphForInsecureEVH(routeIgrObj RouteIngressModel, host, infraSettingName, key string, pathsvcmap HostMetada) {
+	vsNode := o.GetAviEvhVS()
+	ingName := routeIgrObj.GetName()
+	namespace := routeIgrObj.GetNamespace()
+	evhNodeName := lib.GetEvhNodeName(ingName, namespace, host, infraSettingName)
+	evhNode := vsNode[0].GetEvhNodeForName(evhNodeName)
+	hostSlice := []string{host}
+
+	// Populate the hostmap with empty secret for insecure ingress
+	hostMap := HostNamePathSecrets{paths: getPaths(pathsvcmap.ingressHPSvc), secretName: ""}
+	found, ingressHostMap := SharedHostNameLister().Get(host)
+	if found {
+		// Replace the ingress map for this host.
+		ingressHostMap.HostNameMap[namespace+"/"+ingName] = hostMap
+		ingressHostMap.GetIngressesForHostName(host)
+	} else {
+		// Create the map
+		ingressHostMap = NewSecureHostNameMapProp()
+		ingressHostMap.HostNameMap[namespace+"/"+ingName] = hostMap
+	}
+	SharedHostNameLister().Save(host, ingressHostMap)
+
+	if evhNode == nil {
+		evhNode = &AviEvhVsNode{
+			Name:         evhNodeName,
+			VHParentName: vsNode[0].Name,
+			Tenant:       lib.GetTenant(),
+			EVHParent:    false,
+			EvhHostName:  host,
+			ServiceMetadata: avicache.ServiceMetadataObj{
+				NamespaceIngressName: ingressHostMap.GetIngressesForHostName(host),
+				Namespace:            namespace,
+				HostNames:            hostSlice,
+			},
 		}
-		RemoveRedirectHTTPPolicyInModelForEvh(evhNode, hosts, key)
 
-		// build poolgroup and pool
-		aviModel.(*AviObjectGraph).BuildPolicyPGPoolsForEVH(vsNode, evhNode, namespace, ingName, key, infraSettingName, hosts, pathsvcmap.ingressHPSvc)
 		foundEvhModel := FindAndReplaceEvhInModel(evhNode, vsNode, key)
```

Comment:
This is not related to your changes. But FindAndReplaceEvhInModel and following if condition (line #791-794 ) seems redundant... as we are doing same check at line number #813.

Reply:
removed.